### ISSUE TITLE
Readme and examples don't compile

### ISSUE THREAD

**Issue by Pepeye** [streamnative/pulsar-rs#103]
Hi, I'm relatively new to Rust and used Pulsar in other languages (node, golang). Noticed the crate has moved on a bit since i last used it....but the recent updated readme & examples [1c4ef59670015421c790f55fe895c66402c92940] don't compile. Doesn't seem consistent with latest 0.3 version of docs either.

I also updated the tokio import to specify `"... features  = [ "...."] }` to get that working. The rest of the errors below are with respect to the pulsar library only and a direct copy paste of the example code in above commit.

```
error[E0432]: unresolved import `pulsar::TokioExecutor`
 --> src/main.rs:2:32
  |
2 | use pulsar::{Consumer, Pulsar, TokioExecutor,
  |                                ^^^^^^^^^^^^^
  |                                |
  |                                no `TokioExecutor` in the root
  |                                help: a similar name exists in the module: `TaskExecutor`

error[E0107]: wrong number of type arguments: expected 0, found 1
  --> src/main.rs:38:24
   |
38 |     let pulsar: Pulsar<TokioExecutor> = Pulsar::builder(addr).build().await?;
   |                        ^^^^^^^^^^^^^ unexpected type argument

error[E0107]: wrong number of type arguments: expected 0, found 1
  --> src/main.rs:62:25
   |
62 |     let pulsar2: Pulsar<TokioExecutor> = Pulsar::builder(addr).build().await?;
   |                         ^^^^^^^^^^^^^ unexpected type argument

error[E0599]: no function or associated item named `builder` found for struct `pulsar::client::Pulsar` in the current scope
  --> src/main.rs:38:49
   |
38 |     let pulsar: Pulsar<TokioExecutor> = Pulsar::builder(addr).build().await?;
   |                                                 ^^^^^^^ function or associated item not found in `pulsar::client::Pulsar`

error[E0277]: the trait bound `impl futures::future::Future: core::future::future::Future` is not satisfied
  --> src/main.rs:39:24
   |
39 |       let mut producer = pulsar.create_producer("test", Some("my-producer".to_string()), producer::ProducerOptions {
   |  ________________________^
40 | |         schema: Some(proto::Schema {
41 | |             type_: proto::schema::Type::String as i32,
42 | |             ..Default::default()
43 | |         }),
44 | |         ..Default::default()
45 | |     }).await?;
   | |____________^ the trait `core::future::future::Future` is not implemented for `impl futures::future::Future`

error[E0599]: no function or associated item named `builder` found for struct `pulsar::client::Pulsar` in the current scope
  --> src/main.rs:62:50
   |
62 |     let pulsar2: Pulsar<TokioExecutor> = Pulsar::builder(addr).build().await?;
   |                                                  ^^^^^^^ function or associated item not found in `pulsar::client::Pulsar`

error[E0277]: the trait bound `impl futures::future::Future: core::future::future::Future` is not satisfied
  --> src/main.rs:64:44
   |
64 |       let mut consumer: Consumer<TestData> = pulsar2
   |  ____________________________________________^
65 | |         .consumer()
66 | |         .with_topic("test")
67 | |         .with_consumer_name("test_consumer")
...  |
70 | |         .build()
71 | |         .await?;
   | |______________^ the trait `core::future::future::Future` is not implemented for `impl futures::future::Future`

error[E0599]: no method named `try_next` found for struct `pulsar::consumer::Consumer<TestData>` in the current scope
  --> src/main.rs:74:36
   |
74 |     while let Some(msg) = consumer.try_next().await? {
   |                                    ^^^^^^^^ method not found in `pulsar::consumer::Consumer<TestData>`
   |
  ::: /Users/pepeye/.cargo/registry/src/github.com-1ecc6299db9ec823/pulsar-0.3.0/src/consumer.rs:40:1
   |
40 | pub struct Consumer<T: DeserializeMessage> {
   | ------------------------------------------
   | |
   | doesn't satisfy `_: futures_core::stream::TryStream`
   | doesn't satisfy `_: futures_util::stream::try_stream::TryStreamExt`
   |
   = note: the method `try_next` exists but the following trait bounds were not satisfied:
           `pulsar::consumer::Consumer<TestData>: futures_core::stream::TryStream`
           which is required by `pulsar::consumer::Consumer<TestData>: futures_util::stream::try_stream::TryStreamExt`

error[E0599]: no method named `ack` found for struct `pulsar::consumer::Consumer<TestData>` in the current scope
  --> src/main.rs:75:18
   |
75 |         consumer.ack(&msg).await?;
   |                  ^^^ method not found in `pulsar::consumer::Consumer<TestData>`
```

**Comment by Geal** [comment]
hello, it does not compile because the current readme and examples match the next version of pulsar-rs, that is not released yet.

So if you want to use the currently released version (0.3), you can look at older examples: https://github.com/wyyerd/pulsar-rs/tree/8d2a82ebd77e2f94845868cde18ae4e7339dcb3b

Otherwise, you can use the example code if you import it directly from this repository (adding this in Caro.toml: `pulsar = { git = "https://github.com/wyyerdpulsar-rs", rev = "a41cdc5" }`

**Comment by Pepeye** [comment]
Hi @Geal Thanks for getting back to me. I'll wait for the next version then and close the issue in meantime.
